evennia/contrib/unixcommand.py

```diff
@@ -0,0 +1,211 @@
+      It can be used to set options in the parser.  `self.parser`
+      contains the `argparse.ArgumentParser`, so you can add arguments
+      here.
+    - `func`: this method is called to execute the command, but after
+      the parser has checked the arguments given to it are valid.
+      You can access the namespace of valid arguments in `self.opts`
+      at this point.
+
+    The help of UnixCommands is derived from the docstring, in a
+    slightly different way than usual: the first line of the docstring
+    is used to represent the program description (the very short
+    line at the top of the help message).  The other lines below are
+    used as the program's "epilog", displayed below the options.  It
+    means in your docstring, you don't have to write the options.
+    They will be automatically provided by the parser and displayed
+    accordingly.  The `argparse` module provides a default '-h' or
+    '--help' option on the command.  Typing |whelp commandname|n will
+    display the same as |wcommandname -h|n, though this behavior can
+    be changed.
+
+    """
+
+    def __init__(self, **kwargs):
+        super(UnixCommand, self).__init__()
+
+        # Create the empty EvenniaParser, inheriting argparse.ArgumentParser
+        lines = dedent(self.__doc__.strip("\n")).splitlines()
+        description = lines[0].strip()
+        epilog = "\n".join(lines[1:]).strip()
+        self.parser = EvenniaParser(None, description, epilog, command=self)
+
+        # Fill the argument parser
+        self.init()
+
+    def init(self):
+        """
+        Configure the argument parser, adding in options.
+
+        Note:
+            This method is to be overridden in order to add options
+            to the argument parser.  Use `self.parser`, which contains
+            the `argparse.ArgumentParser`.  You can, for instance,
+            use its `add_argument` method.
+
+        """
+        pass
+
+    def func(self):
+        """Override to handle the command execution."""
+        pass
+
+    def get_help(self, caller, cmdset):
+        """
+        Return the help message for this command and this caller.
+
+        Args:
+            caller (Object or Player): the caller asking for help on the command.
+            cmdset (CmdSet): the command set (if you need additional commands).
+
+        Returns:
+            docstring (str): the help text to provide the caller for this command.
+
+        """
+        return self.parser.format_help()
+
+    def parse(self):
+        """
+        Process arguments provided in `self.args`.
+
+        Note:
+            You should not override this method.  Consider overriding
+            `init` instead.
+
+        """
+        try:
+            self.opts = self.parser.parse_args(shlex.split(self.args))
+        except ParseError as err:
+            msg = str(err)
+            if msg:
+                self.msg(msg)
+            raise InterruptCommand
+
+
+class ParseError(Exception):
```

Comment:
This all belongs at the top of the module, before the functions/classes that use it.